python releases broken under python 3.13 and 3.14

With 3.13 and 3.14 uv has to build coincurve, since there a no wheels or anything to pull.

The build system then pulls in the freshly released `scikit-build-core 1.0.0` which turned warning about a variable used by coincurve 20 into an error. Since this is a build dependency it is not included in `uv.lock`. So the only way to really fix this would be new python releases with coincurve pinned to their current master, since only then it would work with 3.13 and 3.14.

```
× Failed to build `coincurve==20.0.0`
  ├─▶ The build backend returned an error
  ╰─▶ Call to `hatchling.build.build_wheel` failed (exit status: 7)

      [stderr]
      ERROR: Use build.verbose instead of cmake.verbose for scikit-build-core >= 0.10

      hint: This usually indicates a problem with the package or the build environment.
  help: `coincurve` (v20.0.0) was included because `cln-meta-project` (v0.1.0) depends on `pyln-proto` (v26.6.2) which depends on `coincurve`
```

Is this the reason why the master branch is broken?
```
FAILED tests/test_reckless.py::test_reckless_uv_install - pyln.client.lightning.RpcError: RPC call failed: method: uvplugintest, payload: {}, error: {'code': -32601, 'message': "Unknown command 'uvplugintest'"}
```

--- a/contrib/pyln-proto/pyproject.toml
+++ b/contrib/pyln-proto/pyproject.toml
@@ -11,10 +11,13 @@ requires-python = ">=3.10,<4.0"
 dependencies = [
     "base58>=2.1.1",
     "bitstring>=4.3.0",
-    "coincurve==20.0.0",
+    "coincurve",
     "cryptography>=46",
     "PySocks>=1"
 ]
+
+[tool.uv.sources]
+coincurve = { git = "https://github.com/ofek/coincurve.git", rev = "7829b29c08ebb1cc80386a1cdaf8c2243c4ef5c5" }
 
 [dependency-groups]
 dev = [

